libdnf/dnf-utils.c

```diff
@@ -141,6 +136,107 @@ dnf_remove_recursive(const gchar *directory, GError **error)
     return TRUE;
 }
 
+
+/**
+ * dnf_file_cleanup:
+ * @src_path: the path to the file
+ * @error: A #GError, or %NULL
+ *
+ * Remove a file based on the file path,
+ * refactored from dnf_remove_recursive() function
+ *
+ * Returns: %FALSE if an error was set
+ *
+ **/
+gboolean
+dnf_file_cleanup(const gchar *src_path, GError **error)
+{
+    if ((unlink(src_path) != 0) && errno != ENOENT) {
+        g_set_error(error,
+                    DNF_ERROR,
+                    DNF_ERROR_INTERNAL_ERROR,
+                    "failed to unlink %s", src_path);
+        return FALSE;
+    }
+
+    return TRUE;
+}
+
+/**
+ * dnf_delete_files_matching:
+ * @path: the top level directory path to look at
+ * @patterns: the patterns that we are expecting from file/directory's name
+ * @error: a #GError instance, to track error
+ *
+ * Remove recursively all the files/directories that have names
+ * which match the patterns. Use with care
+ *
+ * There are several assumptions that are made in this functions:
+ *
+ * 1: We assume the top level path( the path passed in initially)
+ * does not satisfy the criteria of matching the pattern
+ *
+ * 2: We assume the top level path itself is a directory initially
+ *
+ * Since: 0.9.4
+ *
+ **/
+gboolean
+dnf_delete_files_matching(const gchar* directory_path,
+                          const char* const* patterns,
+                          GError **error)
+{
+    const gchar *filename;
+    g_autoptr(GDir) dir = NULL;
+    g_autoptr(GError) error_local = NULL;
+
+    /* try to open the direcotry*/
+    dir = g_dir_open(directory_path, 0, &error_local);
+    if (dir == NULL) {
+        g_set_error(error,
+                    DNF_ERROR,
+                    DNF_ERROR_INTERNAL_ERROR,
+                    "cannot open directory %s: %s",
+                    directory_path, error_local->message);
+        return FALSE;
+    }
+    /* In the directory, we read each file and check if their name matches one of the patterns */
+    while ((filename = g_dir_read_name(dir))) {
+        gchar *src = NULL;
```

Comment:
Needs `g_autofree` and I'd move the assignment together with the declaration, i.e.:
`g_autofree gchar *src = g_build_filename(directory_path, filename, NULL);`

Reply:
Ah, yup, forgot to add the g_autofree. 
>I'd move the assignment together with the declaration

Reason  I did not do it earlier was  because i tend  to follow the same style in this file, fine with changing it :)